### PR TITLE
package resource: Fix `brew` package detection

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
This allows for package detection via `brew` to handle cases where a
particular package formula exists but is not installed.

### CODE REVIEW

**Review by miah — APPROVED** [review]

File: lib/resources/package.rb [hunk 1]
```diff
@@ -194,15 +194,23 @@ class Brew < PkgManagement
...
       pkg = JSON.parse(cmd.stdout)[0]
-      {
-        name: pkg['name'],
-        installed: true,
-        version: pkg['installed'][0]['version'],
-        type: 'brew',
-      }
+      # If package exists but is not installed, then `brew` output will not
+      # contain `pkg['installed'][0]['version']
+      unless pkg.dig('installed', 0, 'version')
```
Comment (adamleff) [inline lib/resources/package.rb]: `unless`...`else` is really frowned-upon in Rubyland.

Let's just make this:

```ruby
return {} unless pkg.dig('installed', 0, 'version')

# the package is installed, so we build up our pkg info hash
{
  name: ...
}
```

**Review by adamleff — CHANGES_REQUESTED** [review]
Great improvement, but we can be more idiomatically correct Ruby-wise.

File: lib/resources/package.rb (hunk 1 above)
Comment (jerryaldrichiii) [reply lib/resources/package.rb]: I like that much more. Implemented in latest commit.

Thanks

**Review by adamleff — APPROVED** [review]

**Review by jquick — APPROVED** [review]
Thanks @jerryaldrichiii !

**Pull request merged**